ui/tests/integration/components/sync/secrets/page/overview-test.js

```diff
@@ -53,135 +53,263 @@ module('Integration | Component | sync | Page::Overview', function (hooks) {
       );
   });
 
-  test('it should render landing cta component for enterprise with the secrets sync feature', async function (assert) {
-    this.destinations = [];
+  test('it should render header, tabs and toolbar for overview state', async function (assert) {
     await this.renderComponent();
 
     assert.dom(title).hasText('Secrets Sync', 'Page title renders');
-    assert.dom(cta.button).hasText('Create first destination', 'CTA action renders');
-    assert.dom(cta.summary).exists('CTA renders');
+    assert.dom(breadcrumb).exists({ count: 1 }, 'Correct number of breadcrumbs render');
+    assert.dom(breadcrumb).includesText('Secrets Sync', 'Top level breadcrumb renders');
+    assert.dom(cta.button).doesNotExist('CTA does not render');
+    assert.dom(tab('Overview')).hasText('Overview', 'Overview tab renders');
+    assert.dom(tab('Destinations')).hasText('Destinations', 'Destinations tab renders');
+    assert.dom(overview.createDestination).hasText('Create new destination', 'Toolbar action renders');
   });
 
-  test('it should render landing cta component for community', async function (assert) {
-    this.version.type = 'community';
-    this.version.features = [];
-    this.destinations = [];
+  module('community', function (hooks) {
+    hooks.beforeEach(function () {
+      this.version.type = 'community';
+      this.version.features = [];
+      this.destinations = [];
+      this.activatedFeatures = [];
+    });
 
-    await this.renderComponent();
+    test('it should show an upsell CTA', async function (assert) {
+      await this.renderComponent();
 
-    assert.dom(title).hasText('Secrets Sync Enterprise feature', 'Page title renders');
-    assert.dom(cta.button).doesNotExist('Create first destination button does not render');
+      assert
+        .dom(title)
+        .hasText('Secrets Sync Enterprise feature', 'page title indicates feature is only for Enterprise');
+      assert.dom(cta.button).doesNotExist();
+    });
   });
 
-  test('it should render landing cta component for enterprise without the secrets sync feature', async function (assert) {
-    this.destinations = [];
-    this.version.features = [];
-    await this.renderComponent();
+  module('enterprise', function (hooks) {
```

Comment:
Labeling the module this way will make it only run on the enterprise test suite, but since this is an integration test and we're already mocking the relevant pieces elsewhere I would suggest maybe labeling this `ent` so it gets run on OSS and we just mock out the pieces the test is reliant on. That way we get full coverage on every PR rather than waiting until the CE -> Ent merge

Reply:
ah, that's super helpful to know. i wondered if that might be the case, thanks for confirming!

i could use more clarity on our testing strategy for OSS vs. enterprise. i keep having to remind myself that some tests are expected to fail if i'm running tests locally with an enterprise binary. let's talk about this during the patterns chat too; i haven't quite figured out exactly what question i'd like to ask here but i suspect a more general chat will help. 😅

Reply:
I'm curious which tests were failing locally with the enterprise binary! If you could mention in our discussion later.

Reply:
@hellobontempo you know, i could've sworn i saw a few integration tests fail locally but now that i have my branch in good shape, the only failure i'm repeatedly seeing locally is this acceptance test:
<img width="1128" alt="Screenshot 2024-03-28 at 9 24 00 PM" src="https://github.com/hashicorp/vault/assets/903288/3971be10-fc80-4115-a3d0-81d2fa14bbbf">

i suppose the failing integration tests i saw earlier was something wrong with my branch, but this test in particular is completely unrelated. 😮